dist/index.js

```diff
@@ -1340,33 +1346,8 @@ async function run() {
 
       // Get pairs by splitting on newlines
       environmentVariables.split('\n').forEach(function (line) {
-        // Trim whitespace
-        const trimmedLine = line.trim();
-        // Skip if empty
-        if (trimmedLine.length === 0) { return; }
-        // Split on =
-        const separatorIdx = trimmedLine.indexOf("=");
-        // If there's nowhere to split
-        if (separatorIdx === -1) {
-            throw new Error(`Cannot parse the environment variable '${trimmedLine}'. Environment variable pairs must be of the form NAME=value.`);
-        }
-        // Build object
-        const variable = {
-          name: trimmedLine.substring(0, separatorIdx),
-          value: trimmedLine.substring(separatorIdx + 1),
-        };
-
-        // Search container definition environment for one matching name
-        const variableDef = containerDef.environment.find((e) => e.name == variable.name);
-        if (variableDef) {
-          // If found, update
-          variableDef.value = variable.value;
-        } else {
-          // Else, create
-          if (variable.value.length !== 0) {
-            containerDef.environment.push(variable); 
-          }
-        }
```

Comment:
I extracted all of this here into two functions (see below), as much of it is re-used.